File: JavaScript/swapTwoNumbers.js
//Program to swap Two numbers without using a third variable

let num1 = 50
let num2 = 10
console.log(`num1 before swap : ${num1}`, `, num2 before swap : ${num2}`);
console.log();

num1 = num1 + num2;
num2 = num1 - num2;
num1 = num1 - num2;

console.log(`num1 after swap : ${num1}`, `, num2 after swap : ${num2}`);
console.log();


/*

Sample case 1 :
  Input => num1 (say 10), num2 (say 20)

  Output =>   num1 before swap : 10, num2 before swap : 20
              num1 after swap : 20,  num2 after swap : 10


Sample case 2 :
  Input => num1 (say -50), num2 (say 10)

  Output =>   num1 before swap : -50, num2 before swap : 10
              num1 after swap : 10,  num2 after swap : -50



  space complexity: O(1)
  time complexity: O(1)
*/

File: Python/Swap-two-numbers.py
# Problem statement 
# write a program to swap two numbers without using third variable 

x = input()
y = input()
  
print ("Before swapping: ") 
print("Value of x : ", x, " and y : ", y)  

x, y = y, x
 
print ("After swapping: ") 
print("Value of x : ", x, " and y : ", y) 

# sample input 
# 10 
# 20

# sample output 
# Before swapping: 
# Value of x :  10  and y :  20
# After swapping: 
# Value of x :  20  and y :  10

# Time complexity : O(1)
# space complexity : O(1)